engine/crates/composition/src/validate.rs

```diff
@@ -68,23 +76,37 @@ fn validate_selection(
         ));
     };
 
+    for required_argument in field.arguments().filter(|arg| arg.r#type().is_required()) {
+        let arg_name = required_argument.name();
+        if selection.arguments.iter().all(|(name, _)| *name != arg_name.id) {
+            ctx.diagnostics.push_fatal(format!(
+                "Error in @requires on {requires_path}: the {field_name}.{arg_name} is required but not provided.",
```

Comment:
should this have the word argument in there?

```suggestion
                "Error in @requires on {requires_path}: the {field_name}.{arg_name} argument is required but not provided.",
```

Reply:
indeed, good catch